cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ArtifactHttpHandlerInternal.java

```diff
@@ -90,6 +105,50 @@ public void listArtifacts(HttpRequest request, HttpResponder responder,
     }
   }
 
+  @GET
+  @Path("/namespaces/{namespace-id}/artifacts/{artifact-name}/versions/{artifact-version}/download")
+  public void getArtifactBytes(HttpRequest request, HttpResponder responder,
+                               @PathParam("namespace-id") String namespaceId,
+                               @PathParam("artifact-name") String artifactName,
+                               @PathParam("artifact-version") String artifactVersion,
+                               @QueryParam("scope") @DefaultValue("user") String scope) throws Exception {
+
+    NamespaceId namespace = validateAndGetScopedNamespace(Ids.namespace(namespaceId), scope);
+    ArtifactId artifactId = new ArtifactId(namespace.getNamespace(), artifactName, artifactVersion);
+    InputStream artifactStream = artifactRepository.newInputStream(Id.Artifact.fromEntityId(artifactId));
+
+    try {
+      responder.sendContent(HttpResponseStatus.OK, new BodyProducer() {
+
+        @Override
+        public ByteBuf nextChunk() throws Exception {
+          ByteBuf buffer = Unpooled.buffer(CHUNK_SIZE);
+          buffer.writeBytes(artifactStream, CHUNK_SIZE);
+          return buffer;
+        }
+
+        @Override
+        public void finished() throws Exception {
+          Closeables.closeQuietly(artifactStream);
+        }
+
+        @Override
+        public void handleError(@Nullable Throwable cause) {
+          LOG.warn(String.format(
```

Comment:
Use placeholder `{}` instead of `String.format` in formatting log message.

Reply:
done